[Switch] Fix OS_Switch::show_virtual_keyboard

hokaze opened the pull request:
Implementation of OS_Switch::show_virtual_keyboard is wrong, arguments list doesn't match the format os.show_virtual_keyboard expects:

os.h
`virtual void show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect = Rect2(), bool p_multiline = false, int p_max_input_length = -1, int p_cursor_start = -1, int p_cursor_end = -1);`

os_switch.h:
`virtual void show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect = Rect2(), int p_max_input_length = -1);`

This is why the show_virtual_keyboard method never executes any of the code within, despite everything else about the inline software keyboard from libnx otherwise largely working.

hokaze committed:
Fixed missing arguments in OS_Switch::show_virtual_keyboard
- was missing the p_multiline, p_cursor_start and p_cursor_end args
(don't urgently need to do anything with those vars, the software keyboard already appears to handle cursor position for editing in the middle of text if that's where you press the touch screen first)
- keyboard height is actually 400, and now reports as such
- removed old printf debugging for hiding software keyboard, it's one of the only places where that remains and shouldn't still be present in release

halotroop2288 commented on platform/switch/os_switch.cpp:
```diff
@@ -456,11 +456,10 @@ void OS_Switch::run() {
 			}
 		}
 
+        swkbdInlineUpdate(&inline_keyboard, NULL);
```
Why did you move this up?

hokaze replied:
Ah, my bad, that was from when I first tested Stary's suggestion from a few months ago of trying to update the keyboard before the input buffer was flushed

As that was originally theorised to be the problem before you spotted the actual issue - I forgot to revert that, ditto for one of my own printf lines

halotroop2288 replied:
Also note the failed style check when you revert this. It needs to be indented with \t not four spaces.

halotroop2288 replied:
You kinda forgot to fix the rest of the indentation but I got it.

halotroop2288 commented on platform/switch/os_switch.cpp:
```diff
@@ -478,15 +477,15 @@ bool OS_Switch::has_virtual_keyboard() const {
-void OS_Switch::show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect, int p_max_input_length) {
-	if (!g_swkbd_open) {
+void OS_Switch::show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect, bool p_multiline, int p_max_input_length, int p_cursor_start, int p_cursor_end) {
+	printf("Showing kbd!\n");
```
This doesn't need a debug print, now that it's fixed.

hokaze replied:
Agreed, will fix.

hokaze committed:
Implemented suggested changes from halotroop2288
- removed remaining printf debug on showing keyboard
- restored swkbdInlineUpdate in main loop to original position after flushing event buffer
(after confirming that running it before is not needed for it to work, like first thought)

hokaze committed:
Code style requires tabs, not spaces
- restored original indentation, previous commits used spaces instead of tabs

halotroop2288 committed:
Update os_switch.cpp

halotroop2288 committed:
Update os_switch.cpp

hokaze commented:
Thanks for doing the cleanup, I somehow completely missed that I hadn't actually removed the printf or the other lines with wrong indents.

I'm taking this as a learning experience for next time, namely to keep in mind the style conventions and not just autopilot there, to review my changes more thoroughly before pushing and to probably do any further coding when more rested and not rocking 2 nights insomnia in a row 😅
